sass/blocks/intro/_intro.scss

```diff
@@ -0,0 +1,23 @@
+.intro {
+  height: 39rem;
+  overflow: hidden;
+  background: url("../images/intro-background.png") center;
+  background-repeat: no-repeat;
+  background-size: cover;
+
+  &-images {
```

Comment:
It looks a bit confusing to me, You have kind of an independent block inside of here. And you we don't break selectors in the middle. 
You have now:
```
.intro
.intro-images
.intro-images__header
```
What's intro-images to intro? it's a separate block. I don't think we need one here. And we nest only child to block component using &. It's not about reusing names, it's about showing hierarchy more and dependence short of.
So I think you should have here something like that:
```
.intro
.intro__images // <- an element that belongs to intro
.intro__header
```